### PR TITLE
support for dynamic tablenames for multi-tenancy #318

### PR DESCRIPTION
**Pull request by bkali**
This is a pull request to implement dynamic table names as described in issue #318

### CODE REVIEW

**Comment by bkali** [comment]
I see that the test coverage % has not passed. I can fix that. Please let me know if you have any other feedback. Thanks,

Biju

**Comment by bkali** [comment]
I added some more test code to increase test coverage.  When I build on my dev machine I see 71% test coverage.  Coveralls is reporting 60.9% test coverage..  I am not sure which one to believe.. :-)

PASS
coverage: 71.4% of statements
ok      github.com/bkali/gorp   103.338s

File: gorp.go [hunk 1]
```diff
@@ -114,6 +114,14 @@ type SqlExecutor interface {
 	queryRow(query string, args ...interface{}) *sql.Row
 }
 
+// DynamicTable allows the users of gorp to dynamically
+// use different database table names during runtime
+// while sharing the same golang struct for in-memory data
+type DynamicTable interface {
+	GetTableName() string
```
Comment (nelsam) [inline gorp.go]: In keeping with [the getters section of Effective Go](https://golang.org/doc/effective_go.html#Getters), can we rename this to `TableName()`?

**Comment by nelsam** [comment]
Sorry about the long delay; I'm going to start going through pull requests gradually and hopefully get things moving toward v2 again.  The only thing I saw that I would prefer changed is the `GetTableName` method, looks good to merge otherwise.

File: gorp.go (hunk 1 above)
Comment (bkali) [inline gorp.go]: Sure, I can change it to TableName(), will update the code within the next couple of days. Thanks!
Biju

**Comment by stxml** [comment]
Weird style for "if" checks. It's the opposite of how standard Go and Gorp uses them, except when you check errors, then it's standard.

```
if nil != foo {
...
}

```
```
if foo != nil {
...
}
```

**Comment by bkali** [comment]
@stxml  : good catch :+1: It is an old habit of mine from C++ days (constants on the left of equality operator to avoid accidental assignment).

**Comment by nelsam** [comment]
I'll just swap the order post-merge :)

I'm going over this one last time and am going to run tests locally.

File: gorp_test.go [hunk 2]
```diff
@@ -404,6 +404,282 @@ type PersistentUser struct {
...
+	// table mapping is correct
+	var dbTenantInst1 = TenantDynamic{curTable: inpInst.curTable}
+	selectSQL1 := "select * from " + inpInst.curTable
+	dbObjs, err := dbmap.Select(&dbTenantInst1, selectSQL1)
+	if err != nil {
+		t.Errorf("Errow in dbmap.Select. SQL: %v, Details: %v", selectSQL1, err)
+	}
+	if nil == dbObjs {
+		t.Errorf("Nil return from dbmap.Select")
+	} else {
```
Comment (nelsam) [inline gorp_test.go]: I'm also updating this - `t.Fatalf` will perform the same comparison but halt the test immediately, making this `else` statement unnecessary.

**Comment by bkali** [comment]
Thank you very much @nelsam !

**Pull request merged**